---
title: "Chr16 SNP Analysis"
author: "Dave Speca"
date: "6/14/2017"
output: html_document
---

```{r}
library(ggplot2)
library(dplyr)
```

###Read in the vcf file and add headers of Chr18.all.vcf:

```{r, cache=TRUE}
# read in .vcf file
Chr16.data <- read.table("Chr16.all.vcf", as.is=TRUE, na.strings=".")
vcf.header <- system("grep '#C' Chr16.all.vcf",intern = TRUE)
vcf.header <- sub("#","",vcf.header) #get rid of the pound sign
vcf.header <- unlist(strsplit(vcf.header,split="\t"))
colnames(Chr16.data) <- vcf.header
head(Chr16.data, 3)
```

###String split data

```{r}
# Before splitting add NAs to blank cells

Chr16.data$DBA[is.na(Chr16.data$DBA)] <- "NA:NA:NA:NA:NA:NA:NA:NA"

DBA.tmp <- matrix(
  unlist(strsplit(Chr16.data$DBA,split = ":")),
  nrow=nrow(Chr16.data),
  byrow=TRUE
  )
head(DBA.tmp)
colnames(DBA.tmp) <- paste("DBA",c("gt","tot.depth","NObsAllele", "ref.depth","ref.qual","alt.depth","alt.qual","gen.lik"),sep="_")

Chr16.data$Chr16[is.na(Chr16.data$Chr16)] <- "NA:NA:NA:NA:NA:NA:NA:NA"

Chr16.tmp <- matrix(
  unlist(strsplit(Chr16.data$Chr16,split = ":")),
  nrow=nrow(Chr16.data),
  byrow=TRUE
  )
head(Chr16.tmp)
colnames(Chr16.tmp) <- paste("Chr16",c("gt","tot.depth","NObsAllele", "ref.depth","ref.qual","alt.depth","alt.qual","gen.lik"),sep="_")

Chr16.data <- cbind(Chr16.data,DBA.tmp,Chr16.tmp,stringsAsFactors=FALSE)
summary(Chr16.data)
```


###Convert columns back to numeric:
```{r}
Chr16.data[,c("DBA_tot.depth","DBA_ref.depth","DBA_ref.qual","DBA_alt.depth","DBA_alt.qual",
            "Chr16_tot.depth","Chr16_ref.depth","Chr16_ref.qual","Chr16_alt.depth","Chr16_alt.qual")] <- 
  apply(Chr16.data[,c("DBA_tot.depth","DBA_ref.depth","DBA_ref.qual","DBA_alt.depth","DBA_alt.qual",
            "Chr16_tot.depth","Chr16_ref.depth","Chr16_ref.qual","Chr16_alt.depth","Chr16_alt.qual")],
        2,
        as.numeric
        )
head(Chr16.data, 4)
```

###Exploratory analysis:

####Sequencing depth: DBA, raw
```{r}
qplot(Chr16.data$DBA_tot.depth, geom="histogram")
```

####Sequencing depth: DBA, focused
```{r}
qplot(Chr16.data$DBA_tot.depth, geom="histogram", xlim=c(20, 100))
```

####Sequencing depth: Chr16, focused

```{r}
qplot(Chr16.data$Chr16_tot.depth, geom="histogram", xlim=c(20, 100))
```

###Scatterplot of DBA and Chr16 sequencing depths (DP)

```{r}
plot1 <- ggplot(Chr16.data, aes(DBA_tot.depth, Chr16_tot.depth))
plot1 + geom_point() + xlim(0,40) + ylim(0,40)
```

###Below 20 reads it gets a little iffy.

###Histogram of QUAL scores:

```{r}
# histogram of low scores
plot2 <- ggplot(Chr16.data, aes(QUAL)) + geom_histogram()
plot2
```

###Well, that's something. Let's try again.

```{r}
# histogram of low scores
plot3 <- ggplot(Chr16.data, aes(QUAL)) + geom_histogram() + xlim(-1, 300)
plot3
```

###Kinda weird that there is a spike at QUAL ~ 60, similar to what was seen in Chr2 and Chr18 data. I don't understand why this is happening.

###Subset Chr16.data QUAL > 40

```{r}
# subset data keeping only SNPs with quality scores greater than or equal to 40
Chr16.Q40.data <- subset(Chr16.data, QUAL >= 40)
```

###Make it a little more user-friendly
```{r}
library(dplyr)
Chr16.Q40.tidy.data <-select(Chr16.Q40.data, -ID, -FILTER, -INFO, -FORMAT, -DBA, -Chr16)
```


###Replot scatterplot of read depths on Q40 dataset:

```{r}
plot4 <- ggplot(Chr16.Q40.data, aes(DBA_tot.depth, Chr16_tot.depth))
plot4 + geom_point() + xlim(20, 600) + ylim(20, 600)
```

###Let's make a table!

```{r}
# count the numbers common and unique to each genotype
ftable(Chr16.Q40.data[,c("DBA_gt","Chr16_gt")])
```

###Legend: 0=reference allele; 1=first alt. allele; 2=second alt. allele; 3=third alt.allele. Therefore, 0/1=a conventional heterozygote.

###The first thing to look at would be DBA-->0/0 (wild type) and Chr16-->1/1 (homozygous mutant)

###But first, define the introgressed region

```{r}
# Subset DBA_gt=="1/1" & Chr16_gt=="1/1"
Chr16.1111.tidy.data <- subset(Chr16.Q40.tidy.data, (DBA_gt=="1/1" & Chr16_gt=="1/1"))
plot.congenic <- ggplot(Chr16.1111.tidy.data, aes(x=POS, y=DBA_tot.depth)) + geom_point(size=0.5) + ylim(0,500)
plot.congenic
```

###Compare this to DBA_gt=="1/1"

```{r}
# Subset DBA_gt=="1/1"
Chr16.11XX.tidy.data <- subset(Chr16.Q40.tidy.data, (DBA_gt=="1/1"))
plot.DBA.SNPs <- ggplot(Chr16.11XX.tidy.data, aes(x=POS, y=DBA_tot.depth)) + geom_point(size=0.5) + ylim(0,500)
plot.DBA.SNPs
```

###Comparing the above two graphs suggests that the introgressed region is ~chr16:30000000-87000000. Let's check the spreadsheet with the markers. BaZing! D16Mit59 (38.4Mb) and D16Mit152 (85.8) were the flanking markers I used for the congenic.

###Subset DBA_gt ==0/0 & Chr2_gt==1/1

```{r}
Chr16.0011.tidy.data <- subset(Chr16.Q40.tidy.data, (DBA_gt=="0/0" & Chr16_gt == "1/1"))
```

####Let's go hunting!

